configfs/fakertmr/fakertmr.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package fakertmr defines a configfsi.Client for faking TSM behavior.
+// The current implementation only supports TDX.
+package fakertmr
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/sha512"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strconv"
+	"syscall"
+
+	"github.com/google/go-configfs-tsm/configfs/configfsi"
+)
+
+const (
+	tsmRtmrDigest = "digest"
+	tsmPathIndex  = "index"
+	tsmPathTcgMap = "tcg_map"
+)
+
+// rtmrValue represents the value of a rtmr index.
+type rtmrValue struct {
+	RtmrIndex int
+	Digest    []byte
+	TcgMap    []byte
+}
+
+// rtmrEntry represents a rtmr entry in the configfs.
+type rtmrEntry struct {
+	RtmrIndex   int
+	Initialized bool
+	// RtmrMaps is a map of rtmr index to rtmr value.
+	// All RrmrMaps must be initialized with the value from RtmrSubsystem.
+	RtmrMaps map[int]*rtmrValue
```

Comment:
I'm not seeing any reason for these fields to be exported when the type isn't. Usually it's for use by reflection libraries for something like a test's use of `cmp.Diff`, but I don't see that here.

Reply:
Make sense. Thank you for the information. Done.